src/main/java/edu/cornell/kfs/sys/service/impl/DocumentMaintenanceServiceImpl.java

```diff
@@ -1,45 +1,108 @@
 
     @Transactional
     public boolean requeueDocuments() {
         Collection<String> docIds = documentMaintenanceDao.getDocumentRequeueValues();
         LOG.info("requeueDocuments: Total number of documents flagged for requeuing: " + docIds.size());
 
+        List<ActionItemNoteDetailDto> noteDetails = documentMaintenanceDao.getActionNotesToBeRequeued();
+        LOG.info("requeueDocuments: Total number of action note details: " + noteDetails.size());
+        
         for (String docId : docIds) {
             LOG.info("requeueDocuments: Requesting requeue for document: " + docId);
-            DocumentRefreshQueue documentRequeuer = KewApiServiceLocator.getDocumentRequeuerService(CoreConfigHelper.getApplicationId(), docId, WAIT_TIME_NO_WAIT);
+            DocumentRefreshQueue documentRequeuer = (DocumentRefreshQueue) SpringContext.getService(CUKFSConstants.DOCUMENT_REFRESH_QUEUE_SERVICE_SPRING_CONTEXT_NAME);
             documentRequeuer.refreshDocument(docId);
+            
+            List<ActionItemNoteDetailDto> noteDetailsForDocument = findNoteDetailsForDocument(noteDetails, docId);
+            for (ActionItemNoteDetailDto detailDto : noteDetailsForDocument) {
+                ActionItem actionItem = findActionItem(detailDto);
+                if (ObjectUtils.isNotNull(actionItem)) {
+                    ActionItemExtension actionItemExtension = findActionItemExtension(actionItem.getId());
+                    if (ObjectUtils.isNull(actionItemExtension)) {
+                        actionItemExtension = buildActionItemExtension(detailDto, actionItem);
+                        LOG.info("requeueDocuments, adding note details " + detailDto.toString() + " to action item " + actionItem.getId());
+                    } else {
+                        actionItemExtension.setActionNote(detailDto.getActionNote());
+                        actionItemExtension.setNoteTimeStamp(detailDto.getNoteTimeStamp());
+                        LOG.info("requeueDocuments, updating note details " + detailDto.toString() + " to action item " + actionItem.getId());
+                    }
+                    KRADServiceLocator.getDataObjectService().save(actionItemExtension);
+                }
+            }
         }
-
         return true;
     }
+    
+    private ActionItem findActionItem(ActionItemNoteDetailDto detailDto) {
+        QueryByCriteria query = QueryByCriteria.Builder.fromPredicates(
+                PredicateFactory.equal(KFSPropertyConstants.PRINCIPAL_ID, detailDto.getPrincipalId()),
+                PredicateFactory.equal(CUKFSConstants.DOCUMENT_ID, detailDto.getDocHeaderId()));
+        List<ActionItem> actionItems = KRADServiceLocator.getDataObjectService().findMatching(ActionItem.class, query).getResults();
+        ActionItem selectedActionItem = null;
+        if (LOG.isDebugEnabled() ) {
+            LOG.info("findActionItem, number of action items for principle " +  detailDto.getPrincipalId() + " and cocument number " 

```

Comment:
There are a couple typos in this message that should be fixed ("principle" and "cocument"). Also, since the block is checking whether debug logging is enabled, I would suggest logging this message at the debug level, unless you meant it to be displayed at the info level (in which case the "if" condition should be changed instead).